### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by alvaroiporre**
## Project requirements
In this PR I made the following:

- Added a column species of type string to your animals table. Modify your schema.sql file.
- Inserted the following data:
  - Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
  - Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
  - Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
  - Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
  - Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
  - Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
  - Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.
  - All these insertions are in the data.sql file.

**Transactions**
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
 ```sh
 BEGIN;
 UPDATE animals set species = 'unspecified';
 SELECT * FROM animals;
 ROLLBACK;
 ```
 Data updated
  Screenshot 2023-07-11 at 18 22 51
 After Rollback
  Screenshot 2023-07-11 at 18 22 28

- Inside a transaction:
  - Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
  - Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
  - Verify that changes were made.
  - Commit the transaction.
  - Verify that changes persist after commit.
  Screenshot 2023-07-11 at 18 27 40

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
  After delete
  Screenshot 2023-07-11 at 18 30 29
  After rollback
  Screenshot 2023-07-11 at 18 27 40

- Inside a transaction:
  - Delete all animals born after Jan 1st, 2022.
  - Create a savepoint for the transaction.
  - Update all animals' weight to be their weight multiplied by -1.
  - Rollback to the savepoint
  - Update all animals' weights that are negative to be their weight multiplied by -1.
  - Commit transaction
  With the error
  Screenshot 2023-07-11 at 18 49 23

  After Rollback to the save point and fix the query
  Screenshot 2023-07-11 at 18 42 19

  - All these transactions are in the queries.sql file.

**Queries**
- Write queries to answer the following questions:
  - How many animals are there?
  Screenshot 2023-07-11 at 18 53 05

  - How many animals have never tried to escape?
  Screenshot 2023-07-11 at 18 55 34

  - What is the average weight of animals?
  Screenshot 2023-07-11 at 18 58 21

  - Who escapes the most, neutered or not neutered animals?
  Screenshot 2023-07-11 at 19 01 01

  - What is the minimum and maximum weight of each type of animal?
  Screenshot 2023-07-11 at 19 03 02

  - What is the average number of escape attempts per animal type of those born between 1990 and 2000?
  Screenshot 2023-07-11 at 19 11 58

  - All these queries are in the queries.sql file.

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -6,5 +6,6 @@ CREATE TABLE animals (
 	escape_attempts int,
 	neutered bool,
 	weight_kg real,
+	species varchar(20),
```
Comment (Lucas-Masaba) [inline schema.sql]: - [ ] Here, it is better to use a command that is going to modify the existing table for example.
`  ALTER TABLE animals ADD species varchar(20);
`
When you have an existing table with data, you would want to modify the table and not make changes that would require you to delete the data.

File: data.sql [hunk 2]
```diff
@@ -5,7 +5,16 @@ INSERT INTO animals (name) VALUES ('Daisy');
...
+VALUES ('Agumon', '2020-02-03', 0, true, 10.23 ),
+('Gabumon', '2018-11-15', 2, true, 8 ),
+('Pikachu', '2021-01-07', 1, false, 15.04 ),
+('Devimon', '2017-05-12', 5, true, 11 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Charmander', '2020-02-08', 0, false, 11 ),
+('Plantmon', '2021-11-15', 2, true, 5.7),
+('Squirtle', '1993-04-02', 3, false, 12.13),
+('Angemon', '2005-06-12', 1, true, 45),
```
Comment (Lucas-Masaba) [inline data.sql]: - [ ] The weight here should be `-45` and not `45` as seen in the example.

File: data.sql [hunk 3]
```diff
@@ -5,7 +5,16 @@ INSERT INTO animals (name) VALUES ('Daisy');
...
-        ('Devimon', '2017-05-12', 5, true, 11 );
+VALUES ('Agumon', '2020-02-03', 0, true, 10.23 ),
+('Gabumon', '2018-11-15', 2, true, 8 ),
+('Pikachu', '2021-01-07', 1, false, 15.04 ),
+('Devimon', '2017-05-12', 5, true, 11 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Charmander', '2020-02-08', 0, false, 11 ),
+('Plantmon', '2021-11-15', 2, true, 5.7),
+('Squirtle', '1993-04-02', 3, false, 12.13),
```
Comment (Lucas-Masaba) [inline data.sql]: - [ ] The weight here should be `-12.13` and not `12.13` as seen in the example.

File: data.sql [hunk 4]
```diff
@@ -5,7 +5,16 @@ INSERT INTO animals (name) VALUES ('Daisy');
...
-        ('Pikachu', '2021-01-07', 1, false, 15.04 ),
-        ('Devimon', '2017-05-12', 5, true, 11 );
+VALUES ('Agumon', '2020-02-03', 0, true, 10.23 ),
+('Gabumon', '2018-11-15', 2, true, 8 ),
+('Pikachu', '2021-01-07', 1, false, 15.04 ),
+('Devimon', '2017-05-12', 5, true, 11 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Charmander', '2020-02-08', 0, false, 11 ),
+('Plantmon', '2021-11-15', 2, true, 5.7),
```
Comment (Lucas-Masaba) [inline data.sql]: - [ ] The weight here should be `-5.7` and not `5.7` as seen in the example.

File: data.sql [hunk 5]
```diff
@@ -5,7 +5,16 @@ INSERT INTO animals (name) VALUES ('Daisy');
...
-        ('Gabumon', '2018-11-15', 2, true, 8 ),
-        ('Pikachu', '2021-01-07', 1, false, 15.04 ),
-        ('Devimon', '2017-05-12', 5, true, 11 );
+VALUES ('Agumon', '2020-02-03', 0, true, 10.23 ),
+('Gabumon', '2018-11-15', 2, true, 8 ),
+('Pikachu', '2021-01-07', 1, false, 15.04 ),
+('Devimon', '2017-05-12', 5, true, 11 );
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
+VALUES ('Charmander', '2020-02-08', 0, false, 11 ),
```
Comment (Lucas-Masaba) [inline data.sql]: - [ ] The weight here should be `-11.0` and not `11.0` as seen in the example.

File: queries.sql [hunk 6]
```diff
@@ -22,3 +22,88 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+/*
+What is the minimum and maximum weight of each type of animal?
+*/
+SELECT MIN(weight_kg), MAX(weight_kg), species FROM animals GROUP BY species;
+
+/*
+What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+*/
+SELECT AVG(escape_Attempts), species FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2020-12-31' GROUP BY species;
```
Comment (Lucas-Masaba) [inline queries.sql]: - [ ] The date here should be `2000/12/31` and not `2020-12-31`

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @alvaroiporre  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Descriptive pull request message :writing_hand: 
- All SQL is in its respective files :file_cabinet: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alvaroiporre** [comment]
Thank you for the feedback I already fixed it.

**Review by Reem-lab — APPROVED** [review]
# Hi @alvaroiporre,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  No Linters errors 🔥
- [ ]  Great job working on the functionality💯
- [ ]  Kudos working on the requested changes🔥
- [ ]  Followed GitFlow ⭐🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by alvaroiporre** [comment]
Thank you for the feedback @Reem-lab.

**Pull request merged**